pkg/syncer/resize_reconciler.go

```diff
@@ -155,7 +155,7 @@ func (rc *resizeReconciler) Run(ctx context.Context, workers int) {
 	stopCh := ctx.Done()
 
 	if !cache.WaitForCacheSync(stopCh, rc.pvcSynced, rc.pvSynced) {
-		log.Errorf("Cannot sync pv/pvc caches")
+		log.Fatal("Cannot sync pv/pvc caches")
```

Comment:
Can you tell me when does this error occur? Currently we do not restart container when we cannot talk to API server or CNS/vSphere, etc.

Reply:
I'm not sure how often this occurs, but the error is when the informer caches dont get populated. 

> Currently we do not restart container when we cannot talk to API server or CNS/vSphere, etc.
Actually we do. If metadata syncer fails to initialize for any reason (cant create k8s client, cant connect to vCenter), we return an error and syncer container restarts - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/cmd/syncer/main.go#L139

But for some reason, we don't follow that in these reconcilers.

Reply:
We should be fixing this.